client/src/app/pages/authorization/sign-in/sign-in.component.ts

```diff
@@ -1,24 +1,33 @@
-import { Component, OnInit, Output, EventEmitter } from '@angular/core';
-import { User } from 'models';
+import { Component, OnInit, OnDestroy, Output, EventEmitter } from "@angular/core";
+import { User } from "models";
 
 @Component({
-  selector: 'app-sign-in',
-  templateUrl: './sign-in.component.html',
-  styleUrls: ['./sign-in.component.css']
+  selector: "app-sign-in",
+  templateUrl: "./sign-in.component.html",
+  styleUrls: ["./sign-in.component.css"]
 })
 export class SignInComponent implements OnInit {
-
-  private _email: string;
+  private _userName: string;
   private _password: string;
 
-  @Output() onSignIn = new EventEmitter<User>();
+  @Output() onSignIn = new EventEmitter<Object>();
+  @Output() onChangeMode = new EventEmitter<User>();
+  constructor() {}
 
-  constructor() { }
+  ngOnInit() {}
 
-  ngOnInit() {
+  handleClick(): void {
+    this.onSignIn.emit({
+      userName: this._userName,
+      password: this._password
+    });
+  
   }
 
-  handleClick(): void{
-    this.onSignIn.emit({nickName: 'sss', email: this._email, password: this._password});
+  changeSignMode(modeType): void {
+    document.getElementById('menu').classList.add("anim2");
```

Comment:
Direct DOM manipulation is bad idea. Here probably you can use angular's animation.